tests/js/parser-server-tests.js

```diff
@@ -0,0 +1,155 @@
+/*
+
+  Test the parser by itself with as little interaction with the validator as possible (only static validator functions used).
+
+  Client-side tests and integration are covered by the validator tests, which make use of the parser internally.
+
+ */
+"use strict";
+var fluid  = fluid || require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var jqUnit = require("jqUnit");
+var path   = require("path");
+var schemaDir = path.resolve(__dirname, "../schemas");
+
+require("../../src/js/common/parser");
+require("../../src/js/common/validate");
+require("../../src/js/server/validate");
+
+// Utility function to wire up and initial wait into each test case.
+require("../../node_modules/gpii-express/tests/js/lib/test-helpers");
+
+fluid.registerNamespace("gpii.schema.parser.tests.server");
+
+gpii.schema.parser.tests.server.testSchemaCaching = function (that) {
+    jqUnit.assertTrue("There should be a dereferenced schema after startup...", Boolean(that.dereferencedSchemas && that.dereferencedSchemas["base.json"]));
+
+    jqUnit.assertTrue("A derived schema should have been correctly dereferenced as well...", Boolean(that.dereferencedSchemas && that.dereferencedSchemas["derived.json"]));
+};
+
+gpii.schema.parser.tests.server.testFieldLookup = function (that, schemaKey, path, expected) {
+    var output = that.lookupField(schemaKey, path);
+    jqUnit.assertDeepEq("The field lookup output should be as expected...", expected, output);
+};
+
+gpii.schema.parser.tests.server.validateAndTest = function (validator, schemaKey, content, expected) {
+    var output = validator.validate(schemaKey, content);
+    jqUnit.assertDeepEq("The output should be as expected...", expected, output);
+};
+
+
+fluid.defaults("gpii.schema.parser.tests.server.caseHolder", {
+    gradeNames: ["gpii.express.tests.caseHolder"],
+    sequenceStart: [
```

Comment:
We should be able to economise on this sequence start material by using a standard testCaseHolder from gpii-express - subject to relying on particularly named events in our own holder